amethyst_controls/src/bundles.rs

```diff
@@ -68,6 +68,7 @@ where
             &[],
         );
         builder.add(MouseCenterLockSystem, "mouse_lock", &["free_rotation"]);
+        builder.add(MouseFocusUpdateSystem::new(), "mouse_focus", &["mouse_lock"]);
```

Comment:
MouseFocusUpdateSystem should be running after FreeRotationSystem but before MouseCenterLockSystem if you want to prevent a cursor jump on focus gained.